### DISCUSSION TITLE
On Navigation Event

### DISCUSSION THREAD

**Discussion by netbizltd** [TanStack/router#283] [Q&A]
**What is the cleanest solution to listen to navigation events?**

Why? For example: It seems to be good practice to check for service worker updates on navigation events to show a "new version available" alert to the user. 

Cheers, Chris

**Comment by convers39** [comment]
try with subscribe(formerly `listen`) API and it works well.

```js
router.history.subscribe(() => {
  console.log('navigate to ', router.history.location.href)
})
```
